network/validator/pubsub/authorized_sender_validator_test.go

```diff
@@ -0,0 +1,293 @@
+	for _, c := range s.unauthorizedMessageOnChannelTestCases {
+		str := fmt.Sprintf("message type (%s) should not be authorized to be sent on channel (%s)", c.MessageStr, c.Channel)
+		s.Run(str, func() {
+			pid, err := unittest.PeerIDFromFlowID(c.Identity)
+			s.Require().NoError(err)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			validate := AuthorizedSenderValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+
+			msgType, err := validate(ctx, pid, c.Message)
+			s.Require().ErrorIs(err, message.ErrUnauthorizedMessageOnChannel)
+			s.Require().Equal(c.MessageStr, msgType)
+
+			validatePubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+			pubsubResult := validatePubsub(ctx, pid, c.Message)
+			s.Require().Equal(pubsub.ValidationReject, pubsubResult)
+		})
+	}
+}
+
+// TestValidatorCallback_ClusterPrefixedChannels checks that the call back returned from AuthorizedSenderValidator correctly
+// handles cluster prefixed channels during validation.
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_ClusterPrefixedChannels() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	identity, _ := unittest.IdentityWithNetworkingKeyFixture(unittest.WithRole(flow.RoleCollection))
+	clusterID := flow.Localnet
+
+	getIdentityFunc := s.getIdentity(identity)
+	pid, err := unittest.PeerIDFromFlowID(identity)
+	s.Require().NoError(err)
+
+	// validate collection consensus cluster
+	validateCollConsensus := AuthorizedSenderValidator(zerolog.Nop(), channels.ConsensusCluster(clusterID), getIdentityFunc)
+	msgType, err := validateCollConsensus(ctx, pid, &messages.ClusterBlockResponse{})
+	s.Require().NoError(err)
+	s.Require().Equal(message.ClusterBlockResponse, msgType)
+
+	validateCollConsensusPubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), channels.ConsensusCluster(clusterID), getIdentityFunc)
+	pubsubResult := validateCollConsensusPubsub(ctx, pid, &messages.ClusterBlockResponse{})
+	s.Require().Equal(pubsub.ValidationAccept, pubsubResult)
+
+	// validate collection sync cluster
+	validateSyncCluster := AuthorizedSenderValidator(zerolog.Nop(), channels.SyncCluster(clusterID), getIdentityFunc)
+	msgType, err = validateSyncCluster(ctx, pid, &messages.SyncRequest{})
+	s.Require().NoError(err)
+	s.Require().Equal(message.SyncRequest, msgType)
+
+	validateSyncClusterPubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), channels.SyncCluster(clusterID), getIdentityFunc)
+	pubsubResult = validateSyncClusterPubsub(ctx, pid, &messages.SyncRequest{})
+	s.Require().Equal(pubsub.ValidationAccept, pubsubResult)
+}
+
+// TestValidatorCallback_ValidationFailure checks that the call back returned from AuthorizedSenderValidator returns the expected validation error.
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_ValidationFailure() {
+	s.Run("sender is ejected", func() {
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
nit: You can just create a single context for the whole test method

Reply:
https://github.com/onflow/flow-go/pull/2648/commits/9086a7e0ae9f81c9a54d847214633fc7b21bb31b